zero_bin/leader/src/http.rs

```diff
@@ -12,20 +12,31 @@ use tracing::{debug, error, info};
 
 /// The main function for the HTTP mode.
 pub(crate) async fn http_main(
-    runtime: Runtime,
+    block_proof_runtime: Runtime,
+    segment_proof_runtime: Runtime,
     port: u16,
     output_dir: PathBuf,
     prover_config: ProverConfig,
 ) -> Result<()> {
     let addr = SocketAddr::from(([0, 0, 0, 0], port));
     debug!("listening on {}", addr);
 
-    let runtime = Arc::new(runtime);
+    let block_proof_runtime = Arc::new(block_proof_runtime);
```

Comment:
In other PR I have put runtime behind `Arc` at the top level to pass it around.

Reply:
should i wait for your PR?

Reply:
Yes will merge it as soon as tests pass